tests/page/page-aria-snapshot.spec.ts

```diff
@@ -15,26 +15,373 @@
+  await page.setContent(`
+    <h1>title</h1>
+    <h1>title 2</h1>
+  `);
+  await checkAndMatchSnapshot(page.locator('body'), `
+    - heading "title"
+    - heading "title 2"
+  `);
+});
+
+it('should snapshot list with accessible name', async ({ page }) => {
+  await page.setContent(`
+    <ul aria-label="my list">
+      <li>one</li>
+      <li>two</li>
+    </ul>
+  `);
+  await checkAndMatchSnapshot(page.locator('body'), `
+    - list "my list":
+      - listitem: one
+      - listitem: two
+  `);
+});
+
+it('should snapshot complex', async ({ page }) => {
+  await page.setContent(`
+    <ul>
+      <li>
+        <a href='about:blank'>link</a>
+      </li>
+    </ul>
+  `);
+  await checkAndMatchSnapshot(page.locator('body'), `
+    - list:
+      - listitem:
+        - link "link"
+  `);
+});
+
+it('should allow text nodes', async ({ page }) => {
+  await page.setContent(`
+    <h1>Microsoft</h1>
+    <div>Open source projects and samples from Microsoft</div>
+  `);
+
+  await checkAndMatchSnapshot(page.locator('body'), `
+    - heading "Microsoft"
+    - text: Open source projects and samples from Microsoft
+  `);
+});
+
+it('should snapshot details visibility', async ({ page }) => {
+  await page.setContent(`
+    <details>
+      <summary>Summary</summary>
+      <div>Details</div>
+    </details>
+  `);
+
+  await checkAndMatchSnapshot(page.locator('body'), `
+    - group: Summary
+  `);
+});
+
+it('should snapshot integration', async ({ page }) => {
+  await page.setContent(`
+    <h1>Microsoft</h1>
+    <div>Open source projects and samples from Microsoft</div>
+    <ul>
+      <li>
+        <details>
+          <summary>
+            Verified
+          </summary>
+          <div>
+            <div>
+              <p>
+                We've verified that the organization <strong>microsoft</strong> controls the domain:
+              </p>
+              <ul>
+                <li class="mb-1">
+                  <strong>opensource.microsoft.com</strong>
+                </li>
+              </ul>
+              <div>
+                <a href="about: blank">Learn more about verified organizations</a>
+              </div>
+            </div>
+          </div>
+        </details>
+      </li>
+      <li>
+        <a href="about:blank">
+          <summary title="Label: GitHub Sponsor">Sponsor</summary>
+        </a>
+      </li>
+    </ul>`);
+
+  await checkAndMatchSnapshot(page.locator('body'), `
+    - heading "Microsoft"
+    - text: Open source projects and samples from Microsoft
+    - list:
+      - listitem:
+        - group: Verified
+      - listitem:
+        - link "Sponsor"
+  `);
+});
+
+it('should support multiline text', async ({ page }) => {
+  await page.setContent(`
+    <p>
+      Line 1
+      Line 2
+      Line 3
+    </p>
+  `);
+
+  await checkAndMatchSnapshot(page.locator('body'), `
+    - paragraph: Line 1 Line 2 Line 3
+  `);
+  await expect(page.locator('body')).toMatchAriaSnapshot(`
+    - paragraph: |
```

Comment:
Supporting multiline here is weird.

Reply:
It is a convenience line wrapping practice in YAML...